master/static/migrations/20230710120538_migrate-checkpointv1.tx.up.sql

```diff
@@ -0,0 +1,110 @@
+-- Endstate of checkpoint views / tables is
+-- raw_checkpoints is left unmodified.
+-- checkpoints_v2 has checkpoint_v2s and unarchived checkpoint_v1s.
+-- checkpoints_view has all unarchived checkpoints pulling from checkpoints_v2.
+-- proto_checkpoints_view is the proto version of checkpoints_view.
+--
+-- Note we just leave checkpoints_v1 data so we can reverse this migration.
+INSERT INTO public.checkpoints_v2 (
+    uuid,
+    task_id,
+    allocation_id,
+    report_time,
+    state,
+    resources,
+    metadata,
+    size
+)
+SELECT
+    -- https://stackoverflow.com/questions/12505158/generating-a-uuid-in-postgres-for-insert-statement
+    COALESCE(c.uuid, uuid_in(overlay(overlay(md5(random()::text || ':' || random()::text) placing '4' from 13) placing to_hex(floor(random()*(11-8+1) + 8)::int)::text from 17)::cstring)),
+    t.task_id,
+    CASE -- TODO is this behaviour okay? Or should we backfill and insert into allocations.
+        WHEN a.allocation_id IS NULL THEN NULL
+        ELSE a.allocation_id
+    END,
+    COALESCE(c.end_time, t.end_time, NOW()),
+    c.state,
+    c.resources,
+    c.metadata,
+    c.size
+FROM public.raw_checkpoints c
+LEFT JOIN trials AS t on c.trial_id = t.id
+LEFT JOIN public.allocations a ON t.task_id || '.' || c.trial_run_id = a.allocation_id
+LEFT JOIN raw_steps AS s ON (
+     -- Hint to the query planner to use the matching index.
+     s.trial_id = t.id
+     AND s.trial_run_id = c.trial_run_id
+     AND s.total_batches = c.total_batches
+)
+LEFT JOIN raw_validations AS v ON (
+    -- Hint to the query planner to use the matching index.
+    v.trial_id = c.trial_id
+    AND v.trial_run_id = c.trial_run_id
+    AND v.total_batches = c.total_batches
+)
+WHERE
+    NOT c.archived AND
+    (s.archived IS NULL OR s.archived = false) AND
+    (v.archived IS NULL OR v.archived = false);
```

Comment:
I think we could probably skip all of this and be fine. When we moved to push arch/core api we got rid of the concept of 'archiving' a checkpoint in the same way we archive out of date steps/validations, i am pretty sure.

Reply:
possibly (though not likely, since we never were supposed to restart before a checkpoint), but i think it is fine.